packages/docusaurus-theme-search-algolia/src/theme/hooks/useSearchQuery.js

```diff
@@ -34,9 +34,8 @@ function useSearchQuery() {
         search: searchParams.toString(),
       });
     },
-    navigateToSearchPage: (searchValue) => {
-      // Refer to https://github.com/facebook/docusaurus/pull/2838
-      history.push(`${baseUrl}search?q=${searchValue}`);
+    generateSearchPageHref: (searchValue) => {
+      return `/search?q=${searchValue}`;
```

Comment:
Using `baseUrl` is required.

Reply:
64060745705377a4dd585c6e31e88abe117183ec